packages/web/components/Setup/SetupProfilePicture.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  Button,
+  Center,
+  Flex,
+  Image as ChakraImage,
+  Input,
+  VStack,
+} from '@metafam/ds';
+import OctoAvatar from 'assets/graphics/octo-avatar.svg';
+import React, { useRef, useState } from 'react';
+
+import { ProfileWizardPane } from './ProfileWizardPane';
+import { WizardPaneCallbackProps } from './WizardPane';
+
+type ImageData = {
+  file: File;
+  width: number;
+  height: number;
+};
+
+export const SetupProfilePicture: React.FC = () => {
+  const field = 'profileImageURL';
+  const [, setImage] = useState<ImageData | null>();
+  const [preview, setPreview] = useState<string | null>();
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  return (
+    <ProfileWizardPane
+      {...{ field }}
+      title="Profile Picture"
+      prompt="Upload an image that will make you instantly recognizable."
+    >
+      {({ register, setter, current }: WizardPaneCallbackProps) => {
+        const { ref: registerRef } = register(field, {});
+
+        const handleUpload = async (e: React.ChangeEvent<HTMLInputElement>) => {
+          const imageFile = e.target.files?.[0];
+          if (!imageFile) return;
+
+          const img = new Image();
+          img.src = URL.createObjectURL(imageFile);
+          img.onload = () => {
+            const imgData = {
+              file: imageFile,
+              width: img.width,
+              height: img.height,
+            };
+            setImage(imgData);
+            setPreview(URL.createObjectURL(imageFile));
+            // need to pass the height and width to the setter in order to pass it to ceramic
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            setter(imgData);
+          };
+        };
+        const handleClick = () => {
+          inputRef.current?.click();
+          inputRef.current?.focus();
+        };
+        const handleRemove = () => {
+          setImage(null);
+          setter('');
+          setPreview(null);
+        };
```

Comment:
If an image is removed and then saved, the edit is discarded, right? Specifically, the image is not removed from Ceramic?

The current UI is a little misleading in that it looks like you can remove an image and save the lack of an image to Ceramic.

The ideal would be to have the removed image persist to Ceramic. I don't know how the schema constraints come into play on this operation.

Barring that, maybe the "Remove" button should reset the image back to its original state. So, a blank image if it's never been set, but the image from the user's profile if it already has a value so it's more obvious you can't remove the image entirely.

This would only be if removing the image can't be persisted to Ceramic.

Reply:
Yes, good catch, setting it as null, and passing null to ceramic works.